mobify/platform-scaffold
# Switch to React Fiber

**JIRA**: [WEB-1238](https://mobify.atlassian.net/browse/WEB-1238)

## Changes
- Update React dependencies to `16.0.0-alpha.3`

> _NOTE: This change requires an update to the SDK to polyfill requestAnimationFrame. Specifically, adding the following to `progressive-web-sdk/scripts/extract-route-regexes.js`:_

    ```
    global.requestAnimationFrame = global.window.requestAnimationFrame || function(fn) {
        return setTimeout(fn, 0)
    }
    ```

## Notes

- [About React Fiber](https://gist.github.com/duivvv/2ba00d413b8ff7bc1fa5a2e51c61ba43)
- [See whether or not React Fiber is actually ready](http://isfiberreadyyet.com/)
- A polyfill for requestAnimationFrame is required (in the SDK) due to [this issue](https://github.com/facebook/react/issues/9102)

## How to test-drive this PR
- Delete your node_modules
- Run `npm install` (new versions of react)
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Verify that the app works without issue

## Impact

Initial tests are showing heavier weight than lighter. Might have some kinks to work out...

~8kb heavier :(

| Before @ `~160kb` | After @ `~168kb` |
| --- | --- |
| ![before-fiber](https://cloud.githubusercontent.com/assets/734535/24378371/8da49a72-12f7-11e7-8517-df5e544953fe.png) | ![after-fiber](https://cloud.githubusercontent.com/assets/734535/24378373/915abc32-12f7-11e7-8ae8-97883433d24c.png) |
